samples/interactive-tutorials/src/main/java/search/tutorials/search_with_facet_spec.md

```diff
@@ -0,0 +1,261 @@
+    ```bash
+    cd java-retail/samples/interactive-tutorials
+    ```
+
+## Getting textual facets, facet_key = "colorFamilies"
+
+To include facets to the search response, you provide a **```FacetSpec```** in
+the search request body, explicitly passing product attributes as facets.
+
+1. In the FacetSpec the only required field is the **```facet_key```**. It
+   defines the product attribute to be used as facet. You can set here ether
+   textual o numerical field name. The full list of the facet keys you can find
+   in the [Retail documentation.](https://cloud.google.com/retail/docs/reference/rpc/google.cloud.retail.v2#facetkey)
+
+2. Now open <walkthrough-editor-select-regex filePath="cloudshell_open/interactive-tutorials/search/SearchWithFacetSpec.java" regex="boost.*0">SearchWithFacetSpec.java</walkthrough-editor-select-regex>.
+
+   In the initial request, the textual facet "colorFamilies" is going to be returned.
+
+   So the SearchRequest is looking like this:
+   ```
+    placement: "projects/945579214386/locations/global/catalogs/default_catalog/placements/default_search"
+    query: "Tee"
+    visitor_id: "123456"
+    page_size: 10
+    facet_specs {
+    facet_key {
+      key: "colorFamilies"
+      }
+    }
+    ```
+
+## Getting textual facets. Result analyse
+
+1. Run the sample in a terminal with the following command:
+    ```bash
+    mvn compile exec:java -Dexec.mainClass="search.SearchWithFacetSpec"
+    ```
+1. Check the response contains the object **```facets```**:
+
+```
+facets {
+  key: "colorFamilies"
+  values {
+    value: "Black"
+    count: 16
+  }
+  values {
+    value: "Blue"
+    count: 8
+  }
+  values {
+    value: "Gray"
+    count: 4
+  }
+  values {
+    value: "Green"
+    count: 14
+  }
+  values {
+    value: "Navy"
+    count: 5
+  }
+  values {
+    value: "Red"
+    count: 4
+  }
+  values {
+    value: "White"
+    count: 4
+  }
+}
+```
+
+## Getting textual facets, facet_key = "brands".
+
+1. Next, change the value of the field **facet_key** and run the code sample again:
+
+```facetKey = "brands"```
+
+1. Now you can check the results. The **facets** object now contains barnds:
+```
+facets {
+  key: "brands"
+  values {
+    value: "#IamRemarkable"
+    count: 3
+  }
+  values {
+    value: "Android"
+    count: 12
+  }
+  values {
+    value: "Google"
+    count: 105
+  }
+  values {
+    value: "Google Cloud"
+    count: 1
+  }
+  values {
+    value: "Stan and Friends"
+    count: 2
+  }
+  values {
+    value: "YouTube"
+    count: 7
+  }
+}
+```
+## Getting numerical facets
+
+To  get the facets for the numerical field, you should specify the intervals of
+values for each faceting field.
+
+Let's modify the ```SearchRequest``` to get price facet with two price intervals: $(0 - 20.0] and $(21.0 - 50]
+
+Define the Interval in the **```getSearchRequest()```**, add the following code:
+```
+    interval1 = Interval()
```

Comment:
where exactly should I pit it in the file?
Is this updated for Java or copied from Python?

Reply:
Updated.